examples/3d-tiles/tileset-3d-layer.js

```diff
@@ -203,43 +249,55 @@ export default class Tileset3DLayer extends CompositeLayer {
           normals: {value: positions, size: 3},
           length: positions.length / 3
         },
-        ...coordinateProps,
         numInstances: pointsCount,
-        getPosition: (object, {index, data, target}) => {
-          target[0] = data.positions[index * 3];
-          target[1] = data.positions[index * 3 + 1];
-          target[2] = data.positions[index * 3 + 2];
-
-          if (transform && coordinateProps.coordinateSystem === COORDINATE_SYSTEM.LNGLAT) {
-            target = new Matrix4().multiplyRight(transform).transformVector3(target);
-            Ellipsoid.WGS84.cartesianToCartographic(target, target);
-          }
-
-          return target;
-        },
-        getColor: (...args) => {
-          return this._getColor(tileHeader, ...args);
-        },
-        getNormal: normals
-          ? (object, {index, data, target}) => {
-              target[0] = data.normals[index * 3];
-              target[1] = data.normals[index * 3 + 1];
-              target[2] = data.normals[index * 3 + 2];
-              return target;
+        ...positionProps,
+        getColor: colors
+          ? (...args) => {
+              return this._getColor(tileHeader, ...args);
             }
-          : [0, 1, 0],
+          : color || [255, 255, 0, 200],
+        getNormal: normals ? (...args) => this._getNormal(...args) : [0, 1, 0],
         opacity: 0.8,
-        pointSize: 1.5
+        pointSize: 1.5,
+        ...transformProps
       })
     );
   }
 
-  /* eslint-disable max-statements */
-  _getColor(tileHeader, object, {index, data, target}) {
-    if (!tileHeader) {
-      return null;
+  _getPosition(tileHeader, object, {index, data, target}) {
+    const {transform} = tileHeader.userData;
```

Comment:
This function should not be needed. The idea is that we should never need to actually change the positions data in the attribute.

We should only need to call `Ellipsoid.WGS84.cartesianToCartographic(target, target)` once, on the coordinate system origin. and make the resulting [lng, lat] our `coordinateSystemOrigin`.

Then we can just provide a `modelMatrix` to the layer that contains all other transforms.